### ISSUE TITLE
Access denied. Project Collection Build Service (REDACTED) needs Use permissions for pool REDACTED

### ISSUE THREAD

**Issue by buckleyGI** [huserben/TfsExtensions#195]
As of yesterday we are seeing the below error without an effective change on our part.

We recreated the stage (it's a classic release) and the problem persists.

What is also a surprise is that our self hosted agent is using version 3.2.2 of the task even though the marketplace _and_  our installation says that 3.1.1 is the latest. How can our agents be using an unreleased version?

We don't think this version mismatch is the root cause however as we saw it operating for a few days and even weeks without the below error message.

![image](https://user-images.githubusercontent.com/2102864/136935106-b36896a1-4368-44ef-bd62-7303d09916b4.png)

![image](https://user-images.githubusercontent.com/2102864/136935116-7e45ff9d-e8a3-4485-8d1f-96e55c30887a.png)

```
Using OAuth Access Token
Provided team project was guid.
Context is Release - using Release Environment Variables
Build shall be triggered for same user that triggered current Release: REDACTED
Source Version: REDACTED
Error message: Error: Access denied. Project Collection Build Service (REDACTED) needs Use permissions for pool REDACTED perform the action. For more information, contact the Azure DevOps Server administrator.
Will wait 1 seconds before retrying request...
Error during request (2/5)
Error message: Error: Access denied. Project Collection Build Service (REDACTED) needs Use permissions for pool Azure Pipelines to perform the action. For more information, contact the Azure DevOps Server administrator.
Will wait 2 seconds before retrying request...
Error during request (3/5)
```

We build up some good experience in debugging DevOps issue but this one baffles us till this point.

**Comment by buckleyGI** [comment]
Hi @huserben

Ok so the version history is a not a mystery anymore. I didn't know it was possibly the diverge from it.

We haven't changed any rights on the pool. Few people have access and worked on the troubleshooting. We also reviewed the audit log for changes in permissions (currently in preview on DevOps).

One thing I haven't mentioned is that the error message talks about the wrong pool:

`Error: Access denied. Project Collection Build Service (REDACTED) needs Use permissions for pool XYZ to perform the action.`

XYZ is not the pool that we expect. We ask the job to be run on pool ABC, names have been changed of course.

For now we have mitigated the issue and are using a PAT instead of OAuth. We think we will continue with the mitigation as the fix. We just have to remember to refresh the PAT on time every year.
